python/pyspark/sql/dataframe.py

```diff
@@ -1272,16 +1272,18 @@ def replace(self, to_replace, value, subset=None):
         """Returns a new :class:`DataFrame` replacing a value with another value.
         :func:`DataFrame.replace` and :func:`DataFrameNaFunctions.replace` are
         aliases of each other.
+        Values `to_replace` and `value` should be homogeneous. Mixed string and numeric
```

Comment:
Challenge accepted :) 

This makes me think we should also document uniqueness requirements. User might expect that:

```
df.replace([1, 1.0], [2, 3.0]) 
```

or 

```
df.replace({1: 2, 1.0: 3.0})
```
when in fact there will be only one pair considered.